Firestore/core/src/firebase/firestore/api/input_validation.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_API_INPUT_VALIDATION_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_API_INPUT_VALIDATION_H_
+
+#include <string>
+
+#include "Firestore/core/src/firebase/firestore/util/string_format.h"
+
+namespace firebase {
+namespace firestore {
+namespace api {
+
+namespace impl {
+
+[[noreturn]] void ThrowIllegalState(const std::string& message);
+[[noreturn]] void ThrowInvalidArgument(const std::string& message);
+
+}  // namespace impl
+
+/**
+ * Throws an exception (or crashes, depending on platform) in response to API
```

Comment:
Optional: I think the comments for the two functions are word-to-word identical. One way to change that would be to:
* make this comment a file comment;
* add new, shorter, comments to each function that exclusively deals with when to use it (i.e., what constitutes an invalid argument and what constitutes illegal state).

Feel free to push back.

Reply:
Done.